src/server/common.h

```diff
@@ -365,21 +365,22 @@ struct ConditionFlag {
 };
 
 // Helper class used to guarantee atomicity between serialization of buckets
-class ConditionGuard {
+class ThreadLocalMutex {
```

Comment:
Consider (perhaps in a follow up PR) to have a member saying which thread ID this was created on, and then in all access (`lock()`, `unlock()` and d'tor) add a `DCHECK()` that it is the same thread. It could find hard to catch misuse bugs.

Reply:
happy to do that, but `get_id` is a member of the `fiber` class and `ThisFiber::` does not expose a method to access directly. Either I need to patch helio or maybe there is a better way to do it. Do you have any suggestions?

Reply:
It is safe to access this mutex from multiple _fibers_. You should check the _thread_ doesn't change (that would be unsafe).
You could do that by, say, comparing `Proactor::tlocal()` (either its id, its address, there are a few ways, all of them would work)

Reply:
oh I thought `get_id` returns the thread id. Right we do this with via proactor ;)